app/controllers/welcome_controller.rb

```diff
@@ -1,5 +1,7 @@
 class WelcomeController < ApplicationController
   def index
-    
+    if session[:user_id]
```

Comment:
I believe this functionality exists in a helper already.  If possible, using that might be good here.